src/components/UserApplyButton/UserApplyButton.tsx

```diff
@@ -19,24 +19,41 @@ interface UserApplyButton {
 
 const UserApplyButton = ({ eventId, eventDetail, applyModalOpen }: UserApplyButton) => {
   const bookmarkRef = useRef<HTMLDivElement>(null);
-  const { isBookmarked } = useIsBookmarkQuery({ eventId });
-  const { eventInfo, formInfo } = eventDetail ?? {};
+  const { isBookmarked } = useIsBookmarkedQuery({ eventId });
+  const { category, hasAlreadyApplied, eventInfo, formInfo } = eventDetail ?? {};
   const { capacity, applicants, isEnded } = eventInfo ?? {};
   const { isAbleToApply } = formInfo ?? {};
 
+  const checkApplyButtonText = () => {
+    if (hasAlreadyApplied) {
+      return EVENT_DETAIL_BUTTON.apply.completed;
+    } else if (category !== 'RECRUITMENT' && applicants >= capacity) {
+      return EVENT_DETAIL_BUTTON.apply.soldOut;
+    } else if (isEnded || isAbleToApply) {
+      return EVENT_DETAIL_BUTTON.apply.deadLine;
+    } else {
+      return EVENT_DETAIL_BUTTON.apply.possible;
+    }
+  };
```

Comment:
p4;
check보단 get의 느낌이 나는 함수인것 같아요!
```suggestion
  const getApplyButtonText = () => {
    if (hasAlreadyApplied) {
      return EVENT_DETAIL_BUTTON.apply.completed;
    } else if (category !== 'RECRUITMENT' && applicants >= capacity) {
      return EVENT_DETAIL_BUTTON.apply.soldOut;
    } else if (isEnded || isAbleToApply) {
      return EVENT_DETAIL_BUTTON.apply.deadLine;
    } else {
      return EVENT_DETAIL_BUTTON.apply.possible;
    }
  };
```